nixos/modules/services/networking/wgautomesh.nix

```diff
@@ -0,0 +1,140 @@
+        options = {
+          publicKey = mkOption {
+            type = types.str;
+            description = mdDoc "Wireguard public key of this peer";
+          };
+          address = mkOption {
+            type = types.str;
+            description = mdDoc "Wireguard address of this peer (a single IP address, multliple addresses or address ranges are not supported)";
+            example = "10.0.0.42";
+          };
+          endpoint = mkOption {
+            type = types.nullOr types.str;
+            description = mdDoc "Bootstrap endpoint for connecting to this Wireguard peer if no other address is known or none are working";
+            example = "wgnode.mydomain.example:51820";
+          };
+        };
+      });
+      description = mdDoc "wgautomesh peer list";
+    };
+  };
+
+  config = mkIf cfg.enable (let
+    peerDefs = map (peer:
+      let
+        endpointDef = if peer.endpoint == null then
+          ""
+        else
+          ''endpoint = "${peer.endpoint}"'';
+      in ''
+        [[peers]]
+        pubkey = "${peer.publicKey}"
+        address = "${peer.address}"
+        ${endpointDef}
+      '') cfg.peers;
+    extraDefs = (if cfg.lanDiscovery then [ "lan_discovery = true" ] else [ ])
+      ++ (if cfg.enableGossipEncryption
+        then [ ''gossip_secret_file = "/var/lib/wgautomesh/gossip_secret"'' ]
+        else [ ])
+      ++ (if cfg.enablePersistence
+        then [ ''persist_file = "/var/lib/wgautomesh/state"'' ]
+        else [ ])
+      ++ (if (cfg.upnpForwardPublicPort != null)
+        then [
+              "upnp_forward_external_port = ${
+                toString cfg.upnpForwardPublicPort
+              }"
+            ]
+        else [ ]);
+    configfile = pkgs.writeText "wgautomesh.toml" ''
+      interface = "${cfg.interface}"
+      gossip_port = ${toString cfg.gossipPort}
+      ${concatStringsSep "\n" extraDefs}
+
+      ${concatStringsSep "\n" peerDefs}
+    '';
+  in {
+    systemd.services.wgautomesh = {
+      enable = true;
+      path = [ pkgs.wireguard-tools ];
+      environment = { RUST_LOG = "wgautomesh=${cfg.logLevel}"; };
+      description = "wgautomesh";
+      serviceConfig = {
+        Type = "simple";
+
+        #ExecStart = "${pkgs.wgautomesh}/bin/wgautomesh ${configfile}";
+        ExecStart = "pkgs.wgautomesh/bin/wgautomesh ${configfile}";
+        Restart = "always";
+        RestartSec = "30";
+
+        ExecStartPre =
+          mkIf cfg.enableGossipEncryption
+          [
+            "+${pkgs.coreutils}/bin/chown wgautomesh /var/lib/wgautomesh/gossip_secret"
+            "+${pkgs.coreutils}/bin/chmod 0600 /var/lib/wgautomesh/gossip_secret"
+          ];
+
+        DynamicUser = true;
+        User = "wgautomesh";
+        StateDirectory = "wgautomesh";
+        StateDirectoryMode = "0700";
+        AmbientCapabilities = "CAP_NET_ADMIN";
+        CapabilityBoundingSet = "CAP_NET_ADMIN";
+      };
+      wantedBy = [ "multi-user.target" ];
```

Comment:
That would be nice but I don't know what the corresponding service is. There is no requirement that the wireguard interface be configured using `networking.wireguard.interfaces`, it could also be a wg-quick service or something else, so the systemd unit name might change.